proxy/proxy.go

```diff
@@ -611,10 +611,10 @@ func (proxy *Proxy) symlink(unixAddrs []string) (err error) {
 	binds := []string{"/var/run/weave:/var/run/weave"}
 	froms := []string{}
 	for _, addr := range unixAddrs {
-		if addr == weaveSockUnix {
+		from := strings.TrimPrefix(addr, "unix://")
+		if from == weaveSock {
```

Comment:
I don't understand how this produces the desired effect - the issue in #2302 is caused by the fact that the supplied `addr` of `/var/run/weave/weave.sock` doesn't match the `weaveSock` values of `unix:///var/run/weave/weave/sock`. Trimming a non-existent `unix://` prefix from such an `addr` doesn't help?